### PR TITLE
Update document for sklearn model IO.

### PR DESCRIPTION
**Pull request by trivialfis**
* Update the use of JSON.
* Remove unnecessary type cast.

### CODE REVIEW

File: python-package/xgboost/sklearn.py [hunk 1]
```diff
@@ -574,9 +548,6 @@ def load_model(self, fname):
             if k == 'classes_':
                 self.classes_ = np.array(v)
                 continue
-            if k == 'use_label_encoder':
-                self.use_label_encoder = bool(v)
-                continue
```
Comment (hcho3) [inline python-package/xgboost/sklearn.py:579]: Why is this line being removed?

File: python-package/xgboost/sklearn.py [hunk 2]
```diff
@@ -542,27 +525,18 @@ def save_model(self, fname: str):
...
 
-        The model is loaded from an XGBoost internal format which is universal
-        among the various XGBoost interfaces. Auxiliary attributes of the
-        Python Booster object (such as feature names) will not be loaded.
-
-        Parameters
-        ----------
-        fname : string
-            Input file name.
-
```
Comment (hcho3) [inline python-package/xgboost/sklearn.py:557]: Can we keep the docstring?

**Review by hcho3 — CHANGES_REQUESTED** [review]

File: python-package/xgboost/sklearn.py (hunk 1 above)
Comment (trivialfis) [reply python-package/xgboost/sklearn.py:579]: We reuse the doc from `booster.load_model`.  See line 563

Comment (trivialfis) [reply python-package/xgboost/sklearn.py:579]: bool value is handled by JSON.

Comment (hcho3) [reply python-package/xgboost/sklearn.py:579]: Given the test, I take your word here.

**Review by hcho3 — APPROVED** [review]

**Review by RAMitchell — APPROVED** [review]

**Pull request merged**